streamalert/alert_processor/outputs/aws.py

```diff
@@ -477,3 +480,180 @@ def _dispatch(self, alert, descriptor):
         LOGGER.info('New Alert:\n%s', json.dumps(publication, indent=2))
 
         return True
+
+@StreamAlertOutput
+class SESOutput(OutputDispatcher):
+    """Handle all alert dispatching for AWS SES"""
+    __service__ = "aws-ses"
+
+    @staticmethod
+    def _add_attachment(msg, name, content):
+        """Add attachments to the msg
+
+        Args:
+            msg (MIMEMultipart): email to attach too
+            name (str): name for the file to be attached
+            content (str): content of the file to be attached (should be string)
+
+        Returns:
+            msg (MIMEMultipart): Email with the relevant attachments
+        """
+        LOGGER.debug("Attaching %s to msg", name)
+
+        att = MIMEApplication(content)
+
+        att.add_header("Content-Disposition", "attachment", filename=name)
+        msg.attach(att)
+
+        return msg
+
+    @staticmethod
+    def _construct_body(msg, body):
+        """ Create the body of the email
+
+        Args:
+            msg (MIMEMultipart): Email Object to contruct body
+            body (str): the body is represented as a string
+            body (dict): dictionary of message_type/message for the body (for use with HTML)
+        """
+        if isinstance(body, str):
+            # For use with string based body
+            LOGGER.debug("body is a string of: %s", body)
+
+            msg.attach(MIMEText(body))
+        elif isinstance(body, dict):
+            # For use with HTML body
+            LOGGER.debug("body is not a string, attaching body of: %s", body)
+
+            textual_message = MIMEMultipart("alternative")
+            for m_type, message in body.items():
+                part = MIMEText(message, m_type)
+                textual_message.attach(part)
+            msg.attach(textual_message)
+
+        return msg
+
+    @classmethod
+    def _build_email(cls, alert, publication, creds):
+        """Construct the email to be sent using the alert, publication and creds
+
+        Args:
+          alert (Alert): The alert
+          publication (dict): Alert relevant to the triggered rule
+          creds (dict): Information relevant to send the alert
+
+        Returns:
+          msg (MIMEMultipart): The constructed email ready to be sent
+        """
+
+        # Presentation defaults
+        default_subject = "{} triggered alert {}".format(
+            alert.rule_name, alert.alert_id
+        )
+        default_body = "Please review the attached record.json"
+
+        # Presentation values
+        subject = publication.get("@aws-ses.subject", default_subject)
+        body = publication.get("@aws-ses.body", default_body)
+
+        msg = MIMEMultipart("mixed")
+
+        # Setup to, from and subject
+        msg["To"] = creds["to_emails"]
+        msg["From"] = creds["from_email"]
+        msg["Subject"] = subject
+
+        # Attach the record to the email
+        if publication.get("@aws-ses.attach_record", True):
```

Comment:
Ohh this is nice. I like this idea--attaching the entire record as an email attachment.

Reply:
It's how I've been using it for a while and makes a lot of sense as default